core/storage/suggestion/gae_models.py

```diff
@@ -334,3 +334,92 @@ def export_data(cls, user_id):
             }
 
         return user_data
+
+
+class GeneralVoiceoverApplicationModel(base_models.BaseModel):
+    """A general model for voiceover application of an entity.
+
+    The ID of the voiceover application will be a random hashed value.
+    """
+    # The type of entity to which the user will be assigned as a voice artist
+    # once the application will get approved.
+    target_type = ndb.StringProperty(required=True, indexed=True)
+    # The ID of the entity to which the application belongs.
+    target_id = ndb.StringProperty(required=True, indexed=True)
+    # The language code for the voiceover audio.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+    # The status of the application. One of: accepted, rejected, in-review.
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # The HTML content written in the given language_code.
+    # This will typically be a snapshot of the content of the initial card of
+    # the target.
+    content = ndb.TextProperty(required=True)
+    # The filename of the voiceover audio. The filename will have
+    # datetime-randomId(length 6)-language_code.mp3 pattern.
+    filename = ndb.StringProperty(required=True, indexed=True)
```

Comment:
Does this need to be indexed?

Reply:
As of now, there isn't any query based on the `filename` but I think this would be useful in the future for GCS data validation or other such cases. (Like we want to make sure whether all the audio files existing in GCS is linked to a model.)

Reply:
I am not sure if that is necessary. @seanlip thoughts? No other concerns from my side

Reply:
It's OK to index this.